javascript/src/components/inputs/InputDropdown.jsx

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+var React = require('react/addons');
+var InputsStore = require('../../stores/inputs/InputsStore');
+var Input = require('react-bootstrap').Input;
+
+var InputDropdown = React.createClass({
+    mixins: [React.addons.LinkedStateMixin],
+    PLACEHOLDER: "placeholder",
+    getInitialState() {
+        return {
+            inputs: [],
+            selectedInput: this.PLACEHOLDER
+        };
+    },
+    loadData() {
+        InputsStore.list((inputs) => {
+            this.setState({inputs: inputs});
+        });
+    },
+    componentDidMount() {
+        if (!this.props.inputs) {
```

Comment:
This is actually only partially correct. 1) & 3) are related. We usually do not need to update the props of this component as the input set is not reloaded live and therefore does not really need a refresh. 2) is not a problem as the map() semantics are identical for maps and arrays when a single argument function is passed.

Nonetheless, as such a small component seems to be hard to understand already, I decided to strip away the loadData() function and all usage of this.state.inputs, so it works only when an input set is passed as a property.